### ISSUE TITLE
CDP SDK Private Key Authentication Failure - Unable to Initialize Agent - Critical Issue Blocking Development

### ISSUE THREAD

**Issue by arhansuba** [coinbase/agentkit#65]
I am encountering persistent authentication failures with the CDP SDK across both the official example code and custom implementations. Despite extensive troubleshooting and following all documentation guidelines, the SDK consistently fails to authenticate with the provided API credentials.

Example Project: CDP Agentkit Chatbot Example

Implementation Attempts:

Official CDP Example Project:

Followed the exact implementation from the CDP documentation
Used the provided example code without modifications
Encountered the same authentication error
Attempted multiple private key formatting approaches

Custom Project Implementation:

Developed according to CDP SDK documentation
Implemented all recommended security practices
Encountered identical authentication failures
Hours of troubleshooting yielded no resolution

Error Message:
'Could not deserialize key data. The data may be in an incorrect format, the provided password may be incorrect, it may be encrypted with an unsupported algorithm, or it may be an unsupported key type (e.g. EC curves with explicit parameters).', [<OpenSSLError(code=503841036, lib=60, reason=524556, reason_text=unsupported)>]
Troubleshooting Steps Taken:

Verified API credentials multiple times in CDP Portal
Attempted various private key formatting approaches
Tested different environment variable configurations
Tried multiple fresh installations
Reviewed all available documentation thoroughly
Spent several days attempting different solutions

Critical Impact:

Complete blockage of development progress
Unable to proceed with any CDP SDK functionality
Both example and custom implementations are non-functional
Significant time lost to troubleshooting

Expected Behavior:
The SDK should successfully authenticate using the provided API credentials and initialize the CDP Agentkit wrapper, allowing development to proceed.
Urgent Resolution Needed:
Given that this issue persists across both official examples and custom implementations, and prevents any development progress, urgent guidance or resolution would be greatly appreciated.

**Comment by murrlincoln** [comment]
Are you using ENV vars for your CDP API Key name / private key?
Here is an example of how the values should be formatted. (Note: I replaced some of the data with sample-value to keep the key secure.)
export CDP_API_KEY_NAME="organizations/sample-value/apiKeys/sample-value"
export CDP_API_KEY_PRIVATE_KEY="-----BEGIN EC PRIVATE KEY-----\nsample-value\n-----END EC PRIVATE KEY-----\n"
Please ensure you are copying the values exactly from cdp_api_key.json file and the env vars are exported as above (wrapped in "…" and \n intact).

Some OS / terminals do not handle the new line literals well when importing / exporting the key as an ENV var. You can pass them as args to the AgentKit constructor if the ENV vars are not working well.


For Python
values = { "cdp_api_key_name": "copy-value-from-cdp-api-key-json", "cdp_api_key_private_key": "copy-value-from-cdp-api-key-json" }
agentkit = CdpAgentkitWrapper(**values)
For Typescript
    const config = {
      cdpApiKeyName: "copy-value-from-cdp-api-key-json",
      cdpApiKeyPrivateKey: "copy-value-from-cdp-api-key-json",
    }

    // Initialize CDP AgentKit
    const agentkit = await CdpAgentkit.configureWithWallet(config);

**Comment by murrlincoln** [comment]
Hey @arhansuba , thanks for being patient with us as we figure this out. 

If you copy and paste the API keys directly from the `cdp_api_key.json` file, you should be able to resolve the issue. Please let me know if this is not the case!

**Comment by mikeghen** [comment]
I was running into the same issue and tried many combinations of ways to load the PK with OS environment variables. The only thing that worked for me was putting the string as it is inline in the file as a kwarg, which bypasses loading it from the ENV. This worked where as loading the key from the ENV did not work. 

TL;DR, got this same error today, work around was to inline the private key as a `cdp_api_key_private_key='....the key string'` which is not really a good way to work around it.

**Comment by its2badwespend** [comment]
7430c4ccc94fd5cc3469ca604169f2d5f55bfa72

**Comment by its2badwespend** [comment]
> 7430c4ccc94fd5cc3469ca604169f2d5f55bfa72

**Comment by mikeghen** [comment]
Loading the key from the OS sometimes leaves `"` around the key, this replace seems to fix that if it happens:
```python
os.getenv("CDP_API_KEY_PRIVATE_KEY").replace('"', '')
```
